scipy/stats/stats.py

```diff
@@ -4401,9 +4407,13 @@ def _perm_test(x, y, stat, compute_distance, reps=1000, workers=-1):
     null_dist : list
         The approximated null distribution.
     """
-    # parallelizes with specified workers over number of reps
+    # parallelizes with specified workers over number of reps and set seeds
     mapwrapper = MapWrapper(workers)
-    parallelp = _ParallelP(x=x, y=y, compute_distance=compute_distance)
+    random_state = check_random_state(random_state)
+    seeds = random_state.permutation(np.arange(reps))
+    random_states = [check_random_state(seeds[i]) for i in range(reps)]
```

Comment:
Suggest using direct iteration: `[check_random_state(seed) for seed in seeds]`

Reply:
`list(map(check_random_state, seeds))` is a less desirable alternative

Reply:
https://github.com/scipy/scipy/pull/11152/commits/32d1e54113332eebbca2dc31d3bb33432a1ae01c